solidity/test/data/redemption.ts

```diff
@@ -0,0 +1,184 @@
+    bitcoinHeaders: BytesLike
+  }
+
+  /**
+   * Chain difficulty which was in force at the moment of Bitcoin transaction
+   * execution. It is used to mock the difficulty provided by `Relay` contract
+   * with a correct value thus making proof validation possible.
+   */
+  chainDifficulty: number
+}
+
+/**
+ * Container for TBTC balance change during redemption test scenario.
+ */
+export interface RedemptionBalanceChange {
+  beforeProof: BigNumber
+  afterProof: BigNumber
+}
+
+/**
+ * `MultiplePendingRequestedRedemptionsWithChange` test data represents a
+ * redemption with the following properties:
+ * - 5 redemption requests
+ * - Redemption dust threshold is 1000000 satoshi
+ * - Treasury fee for each request is 100000 satoshi
+ * - Maximum transaction fee for each request is 1000 satoshi
+ * - Total requested amount for all requests is 6934567 satoshi
+ * - Total treasury fee for all requests is 500000 satoshi
+ * - Total redeemable amount for all requests is 6434567 satoshi
+ * - Maximum total transaction fee is 5000 satoshi
+ */
+export const MultiplePendingRequestedRedemptionsWithChange: RedemptionTestData =
+  {
+    wallet: {
+      pubKeyHash: "0x7ac2d9378a1c47e589dfb8095ca95ed2140d2726",
+      state: 1,
+      pendingRedemptionsValue: 0,
+    },
+
+    redemptionRequests: [
+      {
+        redeemer: "0x7F62CddE8A86328d63B9517BC70B255017f25EEa",
+        redeemerOutputScript:
+          // P2PKH with address mjc2zGWypwpNyDi4ZxGbBNnUA84bfgiwYc
+          "0x1976a9142cd680318747b720d67bf4246eb7403b476adb3488ac",
+        amount: 1000000, // Accepts outputs in range [899000, 900000]
+      },
+      {
+        redeemer: "0x208fF63189DF8749780917Cb5901183075Dbabc1",
+        redeemerOutputScript:
+          // P2WPKH with address tb1qumuaw3exkxdhtut0u85latkqfz4ylgwstkdzsx
+          "0x160014e6f9d74726b19b75f16fe1e9feaec048aa4fa1d0",
+        amount: 1200000, // Accepts outputs in range [1099000, 1100000]
+      },
+      {
+        redeemer: "0x35D54bC29e0a5170c3Ac73E64c7fA539A867f0FE",
+        redeemerOutputScript:
+          // P2SH with address 2MsM67NLa71fHvTUBqNENW15P68nHB2vVXb
+          "0x17a914011beb6fb8499e075a57027fb0a58384f2d3f78487",
+        amount: 2000000, // Accepts outputs in range [1899000, 1900000]
+      },
+      {
+        redeemer: "0x462418b7495561bF2872A0786109A11f5d494aA2",
+        redeemerOutputScript:
+          // P2WSH with address tb1qau95mxzh2249aa3y8exx76ltc2sq0e7kw8hj04936rdcmnynhswqqz02vv
+          "0x220020ef0b4d985752aa5ef6243e4c6f6bebc2a007e7d671ef27d4b1d0db8dcc93bc1c",
+        amount: 1500000, // Accepts outputs in range [1399000, 1400000]
+      },
+      {
+        redeemer: "0x2219eAC966FbC0454C4A2e122717e4429Dd7608F",
+        redeemerOutputScript:
+          // P2PKH with address tb1qgzd40jy62dj5wadvmywfsxugq7f40wzek3p2g2
+          "0x160014409b57c89a53654775acd91c981b88079357b859",
+        amount: 1234567, // Accepts outputs in range [1133567, 1134567]
+      },
+    ],
```

Comment:
There are 5 redemption requests. Each of them pays a `100000` treasury fee. The transaction fee was `4000`. It means each redeemer should contribute `4000 / 5 = 800` sat to the fee.

1000000 - 100000 - 800 = 899200 = 0.008992 BTC
1200000 - 100000 - 800 =  1099200 = 0.010992 BTC
2000000 - 100000 - 800 = 1899200 = 0.018992 BTC
1500000 - 100000 - 800 = 1399200 = 0.013992 BTC
1234567 - 100000 - 800 = 1133767 = 0.01133767 BTC

The wallet had 143565433 sat in the main UTXO. The change should be:
143565433 - 1000000 - 1200000 - 2000000 - 1500000 - 1234567 + 5 * 100000 = 137130866 = 1.37130866

Just line in the TX! 💪🏻 

Maybe we could explain it in the docs for `MultiplePendingRequestedRedemptionsWithChange`? 

I think it nicely illustrates how the system is expected to work.

Reply:
But it is already explained, just in a more concise form.